library/Notifications/Hook/ObjectsRendererHook.php

```diff
@@ -0,0 +1,143 @@
+
+abstract class ObjectsRendererHook
+{
+    /**
+     * Array of Object ID tags for each source
+     *
+     * It has the following structure : ['object source type' => ['object ID' => [object ID tags]]]
+     *
+     * @var array<string, array<string, array<string, string>>>
+     */
+    public static $objectIdTagsCache = [];
+
+    /**
+     * Array of HTMLs for object names with their corresponding object IDs as keys
+     *
+     * It has the following structure : ['object ID' => registered HTML for the object name]
+     *
+     * @var array<string, ValidHtml>
+     */
+    protected static $registeredObjectNameHtmls = [];
+
+    /**
+     * Get the HTML for the object names for the objects using the object ID tags
+     *
+     * @param array<string, array<string, string>> $objectIdTags Array of object ID tags with their corresponding
+     *                                                           object IDs as keys. It has the following structure:
+     *                                                           [''object ID' => [object ID tags]]
+     *
+     * @return array<string, ValidHtml> Array of HTMLs for object names with their object IDs as keys
+     */
+    abstract public function getHtmlForObjectNames(array $objectIdTags): array;
+
+    /**
+     * Get the source type of the objects
+     *
+     * @return string
+     */
+    abstract public function getSourceType(): string;
+
+    /**
+     * Create the object link for the given object ID tag
+     *
+     * @param array<string, string> $objectIdTag
+     *
+     * @return ValidHtml
+     */
+    abstract public function createObjectLink(array $objectIdTag): ValidHtml;
+
+    /**
+     * Register HTMLs to be rendered for the object names using the cached object ID tags
+     *
+     * @return void
+     */
+    final public static function registerHtmlForObjectNames(): void
+    {
+        $objectDisplayNames = [];
+        foreach (Hook::all('Notifications\\ObjectsRenderer') as $hook) {
+            /** @var self $hook */
+            try {
+                $objectDisplayNames[] = $hook->getHtmlForObjectNames(self::$objectIdTagsCache[$hook->getSourceType()]);
+            } catch (Exception $e) {
+                Logger::error('Failed to load hook %s:', get_class($hook), $e);
+            }
+        }
+
+        self::$registeredObjectNameHtmls = array_merge([], ...$objectDisplayNames);
+    }
+
+    /**
+     * Get the registered HTML to be rendered for the object name of the given object
+     *
+     * If an HTML for the object name is not registered, it is created and registered using object ID tags and the same
+     * is returned
+     *
+     * @param Objects $obj
+     *
+     * @return ValidHtml
+     */
+    final public static function getObjectName(Objects $obj): ValidHtml
+    {
+        $objId = $obj->id;
+        if (isset(self::$registeredObjectNameHtmls[$objId])) {
+            return self::$registeredObjectNameHtmls[$objId];
+        }
```

Comment:
And here we can call the new method.
```suggestion
        if (! isset(self::$registeredObjectNameHtmls[$objId])) {
            $data[$obj->source->type][$objId] = $obj->id_tags;
            self::prepare($data);
        }
        
        if (isset(self::$registeredObjectNameHtmls[$objId])) {
            return self::$registeredObjectNameHtmls[$objId];
        }
```

Reply:
I think this is superfluous.